util/strutil/str.go

```diff
@@ -1,9 +1,45 @@
 package strutil
 
-import "unicode"
+import (
+	"strings"
+	"unicode"
+)
+
+const maxInt = int(^uint(0) >> 1)
 
 func CapitalizeFirstLetter(str string) string {
 	var runes = []rune(str)
 	runes = append([]rune{unicode.ToUpper(runes[0])}, runes[1:]...)
 	return string(runes)
 }
+
+func JoinWithTrailingEnd(elems []string, sep string) string {
+	switch len(elems) {
+	case 0:
+		return ""
+	case 1:
+		return elems[0] + sep
+	}
+
+	var n int
```

Comment:
Is this preallocation so necessary? strings.Builder should work pretty well

Reply:
i am simply copied and modified default strings.Join, tried to do less changes, gonna add a comment about it